ambry-replication/src/main/java/com.github.ambry.replication/ReplicaThread.java

```diff
@@ -201,43 +201,23 @@ void replicate(List<List<RemoteReplicaInfo>> replicasToReplicate) {
           startTimeInMs = SystemTime.getInstance().milliseconds();
           fixMissingStoreKeys(connectedChannel, activeReplicasPerNode, exchangeMetadataResponseList);
           fixMissingStoreKeysTimeInMs = SystemTime.getInstance().milliseconds() - startTimeInMs;
-        } catch (Exception e) {
+        } catch (Throwable e) {
           if (checkoutConnectionTimeInMs == -1) {
-            // exception happened in checkout connection phase
+            // throwable happened in checkout connection phase
             checkoutConnectionTimeInMs = SystemTime.getInstance().milliseconds() - startTimeInMs;
-            // recording an exception for any replica on a node will record a node timeout failure
+            // throwable an exception for any replica on a node will record a node timeout failure
```

Comment:
I changed this back to "recording an exception" since `ReplicaHandler` seems to ignore throwables. Should I change the signature (and name) of `ReplicaHandler::onException(ReplicaId, Exception)` to `ReplicaHandler::onThrowable(ReplicaId, Throwable)`? The impl doesn't change of course.